commons/logging/src/main/java/org/operaton/commons/logging/BaseLogger.java

```diff
@@ -276,4 +281,25 @@ protected String exceptionMessage(String id, String messageTemplate, Object... p
     }
   }
 
+  /**
+   *  Sanitize parameters to avoid injection attacks
+   */
+  private Object[] sanitizeParameters(Object... parameters) {
+    if (parameters == null) {
+      return new Object[0];
+    }
+    Object[] sanitized = new Object[parameters.length];
+    for (int i = 0; i < parameters.length; i++) {
+      sanitized[i] = sanitize(parameters[i]);
+    }
+    return sanitized;
+  }
+
+  private Object sanitize(Object parameter) {
+    if (parameter instanceof String) {
+      return ((String) parameter).replaceAll("[\n\r\t]", "_");
```

Comment:
How does sanitizing Newline, Carriage Return and Tab fix security issues? Which vulnerabilities are prevented by this?

Reply:
I was just following the [recommendation from Sonar](https://sonarcloud.io/project/issues?severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&types=VULNERABILITY&pullRequest=411&id=operaton_operaton&open=AZRHr8XIaj5AKEHLTXFF&tab=how_to_fix).

Reply:
Are you aware of some library function that could help here?

Reply:
I dug a little bit on Google about this. What is meant apparently, is that an attacker could smuggle newlines into our logs to lay a false trail while doing something malicious, so banning these characters would prevent this. The suggested fix by Sonar addresses this "log forging".

XSS or Log Injection with variables like log4shell should be handled by the logging framework, so this seems to be fine, and the fix would make sense.

Reply:
For future discussion (not on this issue, maybe): https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html